File: backend/Pubquiz.Domain/ResultCode.cs
namespace Pubquiz.Domain
{
    public enum ResultCode
    {
        /* Error codes */
        InvalidCode,
        TeamNameIsTaken,
        InvalidEntityId,
        QuestionNotInQuiz,
        QuestionNotInCurrentQuizSection,
        InvalidGameStateTransition,
        InvalidUserNameOrPassword,
        QuizMasterUnauthorizedForGame,
        QuizMasterUnauthorizedForQuizItem,
        TeamCantAccessQuizItemOtherThanTheCurrent,
        LobbyUnavailableBecauseOfGameState,
        NoRoleClaimForUser,
        UnauthorizedRole,
        ValidationError,
        GameIsPausedOrFinished,

        /* Success codes */
        Ok,
        ThatsYou,
        LoggedOut
    }
}

File: team-frontend/src/models/ResultCode.ts
export enum ResultCode {
    /* Error codes */
    InvalidCode = 'InvalidCode',
    TeamNameIsTaken = 'TeamNameIsTaken',
    InvalidTeamId = 'InvalidTeamId',
    InvalidQuestionId = 'InvalidQuestionId',
    QuestionNotInQuiz = 'QuestionNotInQuiz',
    QuestionNotInCurrentQuizSection = 'QuestionNotInCurrentQuizSection',
    InvalidInteractionId = 'InvalidInteractionId',
    InvalidGameStateTransition = 'InvalidGameStateTransition',
    InvalidGameId = 'InvalidGameId',
    InvalidUserNameOrPassword = 'InvalidUserNameOrPassword',
    InvalidUserId = 'InvalidUserId',
    QuizMasterUnauthorizedForGame = 'QuizMasterUnauthorizedForGame',
    LobbyUnavailableBecauseOfGameState = 'LobbyUnavailableBecauseOfGameState',
    NoRoleClaimForUser = 'NoRoleClaimForUser',
    NoCurrentGameIdClaimForUser = 'NoCurrentGameIdClaimForUser',
    UnauthorizedRole = 'UnauthorizedRole',
    TeamAlreadyLoggedIn = 'TeamAlreadyLoggedIn',
    InvalidItemNavigation = 'InvalidItemNavigation',
    ValidationError = 'ValidationError',
    /* Success codes */
    Ok = 'Ok',
    ThatsYou = 'ThatsYou',
    LoggedOut = 'LoggedOut'
}
